### ISSUE TITLE
Label renders out of its parent with RTL language

### ISSUE THREAD

**Issue by mahdi739** [vizia/vizia#335]
I created a simple button and a label within it.
Then I gave the label a width so I expected button to become as wide as its child.
It worked as expected for English text but I wrote in an RTL language and it rendered out of the button.
` Application::new(|cx| {
        Button::new(
            cx,
            |cx| cx.emit(AppEvent::Decrement),
            |cx| Label::new(cx, "سلام").width(Pixels(100.0)),
        );
    })
    .run();`
Is it an expected behavior?

**Comment by geom3trik** [comment]
I don't think this is expected behaviour in this case, but is it possible that the text is just too long for the 100px width of the label? Are you able to add a screenshot of the result?

**Comment by mahdi739** [comment]
No. I wrote the same text "سلام"
![Screenshot 2023-04-12 011036](https://user-images.githubusercontent.com/86552031/231283667-9a7bec29-4bb5-4d5b-9b05-de5fe641c54f.jpg)

**Comment by geom3trik** [comment]
Okay I've dug into this and found that it happens because cosmic-text, which we use for text rendering, aligns rtl text to the right of its container by default, which I didn't realise so it conflicts with my own code which attempts to align the text within the container.

A newer version of cosmic-text added the ability to specify the alignment of text within a container so I will update vizia to account for this. However, this will likely end up as part of some bigger changes I am making to text layout/rendering in this PR: #291, which hopefully shouldn't take too long now to finish and merge.
